pkg/background/request_process.go

```diff
@@ -21,3 +25,29 @@ func (c *Controller) ProcessUR(ur *urkyverno.UpdateRequest) error {
 	}
 	return nil
 }
+
+func (c *Controller) MarkUR(ur *urkyverno.UpdateRequest) (bool, error) {
+	handler := ur.Status.Handler
+	if handler != "" {
+		if handler != config.KyvernoPodName {
+			return false, nil
+		}
+		return true, nil
+	}
+
+	handler = config.KyvernoPodName
+	ur.Status.Handler = handler
+	_, err := c.kyvernoClient.KyvernoV1beta1().UpdateRequests(config.KyvernoNamespace).UpdateStatus(context.TODO(), ur, metav1.UpdateOptions{})
+	return true, err
+}
+
+func (c *Controller) UnmarkUR(ur *urkyverno.UpdateRequest) error {
+	newUR, err := c.urLister.Get(ur.Name)
+	if err != nil {
+		return err
+	}
+
+	newUR.Status.Handler = ""
```

Comment:
Should we check if handler is `" "` before updating ?

Reply:
`UnmarkUR` is invoked after a UR is marked and processed by the current instance. This is to release the processed UR which must have a handler set.